user/items_defenition.py

```diff
@@ -1,90 +1,125 @@
-from collections import Counter
 import items.itemloader as itemloader
+from items.item_info import item_info
+
+def _remove_item(item, ind, count=1):
+	'''
+	Removes count of item.
+	If this item completely removed (count <= 0), returns ind. Otherwise returns -1
+	'''
+	if item.is_simple():
+		new_count = item.count - count
+		if new_count <= 0:
+			return ind
+		else:
+			item.count = new_count
+	else:
+		return ind
+	return -1
+
+def _find_item_by_name(items, name):
+	for i, item in enumerate(items):
+		if item.name == name:
+			return i, item
+
+	return None
+
+def _remove_item_by_name(info, items, name):
+	'''
+	Searches for item with given name (not code_name) and removes it.
+	info -- source of item_infos
+	items -- source of loaded items
+	'''
+	found = _find_item_by_name(items, name)
+	if found is None:
+		return False
+	ind, item = found
+	ind = _remove_item(info[ind], ind)
+	if ind >= 0 and items[ind] and not items[ind].iscursed:
+		del info[ind]
+	return True
 
-def remove_item(self, code_name):
+def remove_item(self, code_name, count=1):
```

Comment:
if we will have some item of one code_name with and without context it won't work right